src/utils/utils.hpp

```diff
@@ -20,34 +20,23 @@ namespace mg_memory {
 /// Unique pointers with custom deleters for automatic memory management of
 /// mg_values.
 
-template <class T>
-inline void CustomDelete(T *);
+template <class T> inline void CustomDelete(T *);
 
-template <>
-inline void CustomDelete(mg_session *session) {
+template <> inline void CustomDelete(mg_session *session) {
```

Comment:
2 things here -- even though it's not your code and the changes are from the formatter --  it's worth mentioning.
1. Never ever specialize function templates. Overload instead. The reason is that overload resolution, never considers specializations but only the primary candidate vs its overload. If and only if, the primary template is a better match, then the overload resolution continues and searches if there exists a specialization that is a better match. 
2. Inline is not needed. templates are implicitly inlined.

Reply:
Thanks for mentioning this :smile: will remove inline, and see if there is something to do about removing templates as well

Reply:
ah the 2. point is only true for unspecialized templates, the ones that are specialized need inline...